public/pages/workflows/workflows.tsx

```diff
@@ -116,12 +116,15 @@ export function Workflows(props: WorkflowsProps) {
   // If the user navigates back to the manage tab, re-fetch workflows
   useEffect(() => {
     if (selectedTabId === WORKFLOWS_TAB.MANAGE) {
-      dispatch(
-        searchWorkflows({
-          apiBody: FETCH_ALL_QUERY,
-          dataSourceId: dataSourceId,
-        })
-      );
+      // wait until selected data source is ready before doing dispatch calls if mds is enabled
+      if (!dataSourceEnabled || (dataSourceId && dataSourceId !== '')) {
```

Comment:
nit: lets move this boolean into a constant for readability and since it's used in other places.

Reply:
This hook will only be triggered on changes to `selectedTabId`, hence if the data source ID comes in async later, this won't be re-run. Suggest to add these dependent vars into the list of listening vars alongside `selectedTabId` on line 129.

Reply:
Tyler, this case is handled here https://github.com/opensearch-project/dashboards-flow-framework/pull/530/files#diff-a3a1d6b511eaf420acbca3b67675dae82fd466bac8d2a8cf5f3e2aee7392424fR158